Airlock Seals 2: I'm dumb and a fool edition

Jimmius opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
This is basically just an exact remake of #52595 except without all the bumbling fuckups I made due to being dumb. Mega thanks to Rohesie for helping me, considering literally all of the problems were ones I made for myself (the merge conflict was literally due to a missing cricket bat icon). In conclusion, I'm dumb.

> Adds door seals to the game, inspired by the ones used in Alien Isolation and SOMA. Trivially easy for a human to remove, but makes it a nightmare to try and smash through if you're an alien, an evil robot, or don't have hands. I also slightly changed how airlock reinforcement works, as now there are two ways to increase the integrity, the previous way of just resetting the max_integrity after the removal of the plating wouldn't work.

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
Makes it harder for aliens, cyborgs, and simplemobs to get into places the crew don't want them, while being easy to remove by humans.
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl:
add: Adds door seals to the game, printable at engineering and security lathes after they've been researched. Makes airlocks much harder to smash down if you can't remove them.
tweak: Slightly changed how bonus health to an airlock works
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Commit:
Merge pull request #1 from tgstation/master

cool pull request

Commit:
Merge branch 'master' of https://github.com/Jimmius/tgstation

Commit:
Adds Door Seals

I'll be amazed if this actually works, but lets give it a punt

Commit:
oops

You can no longer infinitely remove seals to weaken airlocks

Commit:
adds the concept of time

please work

Commit:
hopefully adds documentation unless my brain is smaller than I think

I really hope so

Commit:
makes the proc actually return a value if there's no seal

WHOOPS

Commit:
finally adds documentation for the proc

Also changes the return from a failed attempt to TRUE, so a failed attempt returns correctly and we don't continue attempting to open the door (previously this would trigger the safety lock on an unpowered arrivals door despite it being sealed)

Commit:
adds the seal overlay sprite to a few more overlay files I forgot about

classic

Commit:
adds the argument to the proc documentation

I guess

Commit:
changes a user message to be more consistent

basically just changes "this airlock" to "[src]"

Commit:
hopefully adds more early returns unless I forgot one somewhere, also maybe nulls the variables, if I understood what that means correctly

Commit:
forgot to change a variable for an early return

oops

Commit:
removes a thing that I'm not sure what it does

Commit:
Adds the variable null stuff back in since that's not what's causing appveyor and travis to shit themselves

also adds a thing to stop message spam if you unseal constantly

Commit:
maybe more whitespace is the answer!

Commit:
alright let's just delete the attack verbs and see what happens

Commit:
the seal now blocks RCDs to make athah happy

if you think it shouldn't then yell at me and I'll undo it

Commit:
Merge branch 'master' of https://github.com/tgstation/tgstation

Commit:
Merge branch 'master' into i-honestly-have-no-idea

Commit:
adds the icon

Commit:
Merge branch 'master' of https://github.com/Jimmius/tgstation

Commit:
Merge branch 'master' into i-honestly-have-no-idea

Commit:
attempt to fix the merge conflict

actioninja commented:
I forgot to ever comment again but yeh I actually just completely misunderstood the feature, it actually seems fine.

Commit:
Merge branch 'master' of https://github.com/Jimmius/tgstation

Commit:
Merge branch 'master' into i-honestly-have-no-idea

Commit:
attempt to resolve second merge conflict

cacogen commented:
This will make me play Alien: Isolation finally. Fuck aliens and fuck simple mobs.

Commit:
Merge branch 'master' of https://github.com/Jimmius/tgstation

Commit:
Merge branch 'master' into i-honestly-have-no-idea

Commit:
I love fixing merge conflicts, I just love it

I don't actually love it, please merge this so I can forget about it

anconfuzedrock commented:
wait a second aren't airlock seals basically reinforced tables
gotta be human to climb a table

Jimmius commented:
sort of, a lot of the things this is supposed to counter can just walk over tables anyway, like spiders and xenos and what have you. It also doubles the integrity of the airlock, which welding doesn't do.

Commit:
Merge branch 'master' of https://github.com/Jimmius/tgstation

Commit:
Merge branch 'master' into i-honestly-have-no-idea

Commit:
Merge Conflict IV: the Conflictening

in cinemas now

Timberpoes commented on code/game/machinery/doors/airlock.dm:
```diff
@@ -70,6 +71,7 @@
 	var/obj/item/electronics/airlock/electronics
 	COOLDOWN_DECLARE(shockCooldown)
 	var/obj/item/note //Any papers pinned to the airlock
+	var/obj/item/seal //The seal on the airlock
```
This needs to be documented as per our [documentation guidelines](https://github.com/tgstation/tgstation/blob/master/.github/AUTODOC_GUIDE.md)

Jimmius replied:
I  think  that's resolved now

Timberpoes commented on code/game/machinery/doors/airlock.dm:
```diff
@@ -34,6 +34,7 @@
 
 #define AIRLOCK_INTEGRITY_N			 300 // Normal airlock integrity
 #define AIRLOCK_INTEGRITY_MULTIPLIER 1.5 // How much reinforced doors health increases
+#define AIRLOCK_SEAL_MULTIPLIER		 2 // How much extra health airlocks get when braced with a seal
```
This needs to be documented as per our [documentation guidelines](https://github.com/tgstation/tgstation/blob/master/.github/AUTODOC_GUIDE.md)

Jimmius replied:
I think this is resolved now

Timberpoes commented on code/game/machinery/doors/door.dm:
```diff
@@ -97,6 +98,10 @@
 			return TRUE
 	return FALSE
 
+/// Triggered when attempting to remove an airlock seal
+/obj/machinery/door/proc/try_remove_seal(mob/user)
+	return
```
Could use improved documentation as per our [documentation guidelines](https://github.com/tgstation/tgstation/blob/master/.github/AUTODOC_GUIDE.md)

Please add information on the user argument for try_remove_seal. Doesn't have to be long, as long as it's documented as per the guidelines.

Jimmius replied:
I think that's resolved

Timberpoes commented:
Geez laweez. Sorry this has been hanging so long. I've now assigned myself to it.

I'll see about getting this merged as soon as. It's definitely overdue and I'm not sure how you slipped under our radars.

Current triage is as follows:

- [ ] Code review -> Minor notes requesting improved documentation.
- [ ] Icon review -> I'll try and poke @Twaticus or someone else to review the new icons and sprite work.
- [ ] Have you tested this? -> While the above is happening, you can help expedite the process by explaining any testing you've done to be sure this feature works as intended.
- [ ] Mergies -> When the above 3 steps are completed, we'll get this in.

Commit:
Correct documentation (I think), also changes the item sprite

so it matches the overlay on the door better (i made it  less rectangular by a single pixel)

Jimmius commented:
Things  wot I have tested so far:

1: I have used  the seal on every single  airlock currently  in the game. It works and displays correctly, and I can remove it from all of them. 
2: Non-Cultists can safely remove this from Cult airlocks without getting bounced, so culties don't get a free lunch.
3: If an airlock with safety unlocks is unpowered and sealed, clicking will remove the seal (you can unlock the safety system once it's removed).
4: The seal keeps the door closed during emags and staff of doors, and blocks attempts to open from the AI and Cyborgs, as well as pulsing the open wire.
5: Everything works fine if you seal AND reinforce and airlock, in either order.

If there's any other edge case interactions that I can't think of, just say and I'll test them.

Timberpoes commented on code/game/machinery/doors/door.dm:
```diff
@@ -98,7 +98,14 @@
+  * Called when attempting to remove the seal from an airlock
+  *
+  * Here because we need to call it and return if there was a seal so we don't try to open the door
+  * or try it's safety lock while it's sealed
```
```suggestion
  * or try its safety lock while it's sealed
```

Timberpoes commented on code/game/machinery/doors/door.dm:
```diff
@@ -98,7 +98,14 @@
+  * Called when attempting to remove the seal from an airlock
+  *
+  * Here because we need to call it and return if there was a seal so we don't try to open the door
+  * or try it's safety lock while it's sealed
```
```suggestion
  * or try its safety lock while it's sealed
```

Timberpoes replied:
Yes, I am being that pedantic! Muahahaha.

He, his. Her, hers. It, its.

Jimmius replied:
fuck, how did I not notice this

Commit:
its its its its its its

its its its its its
